### ISSUE TITLE
Building subset of Scala versions and/or platforms

### ISSUE THREAD

**Issue by sideeffffect** [oyvindberg/bleep#204]
These are surprisingly common use cases:
 * compile all projects that have Scala version 2.13.*
 * test all projects that have Scala version 2.12.* and platform JVM
 * test all projects that have Scala platform Native
 * etc...

Even though it is unusual to want this, other tools can't do this well, not even sbt.

What is Bleep's story for this?

**Comment by oyvindberg** [comment]
Much, much better than sbt IMO, and I think we can make it even better.

Whenever you can provide cross project names to bleep, for instance to `bleep compile/test/publish-local` you can also provide a `crossId` instead, which will select all cross projects with that `crossId`. 
If you have imported a cross-built project you will have seen names like `myproject@jvm213` - the `crossId` is the part after `@`.

These are somewhat discoverable also because they appear in tab-completions, if you have installed those.

todo:
- document `crossId`s. I'm currently trying to describe the model, templates, cross projects and so on. it's difficult work which brings to light some imperfections, so I hope to have it ready within a few weeks.
- I think "native", "213" could also be added to select by only platform or scala version, try to add that as well (see ProjectGlobs.scala)
- document how to install tab completions (or even better, provide a command which will edit `~/.profile` automatically)

And then to the coolest thing: This is also supported in IDEs!
I'll just demo it:

https://user-images.githubusercontent.com/247937/199306031-78e5d206-cd4c-427d-9a82-82fbbfc4520b.mp4

**Comment by oyvindberg** [comment]
tell me if you want to handle one of the two latter points @sideeffffect

**Comment by sideeffffect** [comment]
So

> compile all projects that have Scala version 2.13.*
```
bleep compile 213
```
> test all projects that have Scala version 2.12.* and platform JVM
```
bleep test jvm212
```
> test all projects that have Scala platform Native
```
bleep test native
```
> publish locally all projects that have Scala version 3.*
```
bleep publishLocal 3
```

Did I get it right?

**Comment by oyvindberg** [comment]
Yes basically. I'm not really happy with just `3` and `native`. `jvm213` works for me since it's longer and more self explaining. Thinking some kind of naming pattern for all the groups of projects would be better. Ideas?

Also by the way, if you want to list which projects end up in a certain group you can say `bleep projects jvm213` for instance
